model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/authorization/policy/entity/JpaPolicyEntity.java

```diff
@@ -0,0 +1,365 @@
+    @Override
+    public void setResourceServerId(String resourceServerId) {
+        metadata.setResourceServerId(resourceServerId);
+    }
+
+    @Override
+    public String getName() {
+        if (isMetadataInitialized()) return metadata.getName();
+        return name;
+    }
+
+    @Override
+    public void setName(String name) {
+        metadata.setName(name);
+    }
+
+    @Override
+    public String getType() {
+        if (isMetadataInitialized()) return metadata.getType();
+        return type;
+    }
+
+    @Override
+    public void setType(String type) {
+        metadata.setType(type);
+    }
+
+    @Override
+    public String getDescription() {
+        return metadata.getDescription();
+    }
+
+    @Override
+    public void setDescription(String description) {
+        metadata.setDescription(description);
+    }
+
+    @Override
+    public DecisionStrategy getDecisionStrategy() {
+        return metadata.getDecisionStrategy();
+    }
+
+    @Override
+    public void setDecisionStrategy(DecisionStrategy decisionStrategy) {
+        metadata.setDecisionStrategy(decisionStrategy);
+    }
+
+    @Override
+    public Logic getLogic() {
+        return metadata.getLogic();
+    }
+
+    @Override
+    public void setLogic(Logic logic) {
+        metadata.setLogic(logic);
+    }
+
+    @Override
+    public Map<String, String> getConfigs() {
+        return Collections.unmodifiableMap(config.stream().collect(Collectors.toMap(JpaPolicyConfigEntity::getName, JpaPolicyConfigEntity::getValue)));
+    }
+
+    @Override
+    public void setConfigs(Map<String, String> config) {
+        this.config.clear();
+        if (config == null) return;
+        for (Map.Entry<String, String> entry : config.entrySet()) {
+            setConfig(entry.getKey(), entry.getValue());
+        }
+    }
+
+    @Override
+    public String getConfig(String name) {
+        return config.stream()
+                .filter(obj -> Objects.equals(obj.getName(), name))
+                .findFirst()
+                .map(JpaPolicyConfigEntity::getValue)
+                .orElse(null);
+    }
+
+    @Override
+    public void setConfig(String name, String value) {
+        if (name == null || value == null || value.trim().isEmpty()) return;
+        JpaPolicyConfigEntity configEntity = new JpaPolicyConfigEntity(this, name, value);
+        if (config.contains(configEntity)) config.remove(configEntity);
+        config.add(configEntity);
+    }
+
+    @Override
+    public void removeConfig(String name) {
+        config.removeIf(obj -> Objects.equals(obj.getName(), name));
+    }
+
+    @Override
+    public Set<String> getAssociatedPolicyIds() {
+        return policyIds.stream().map(UUID::toString).collect(Collectors.toSet());
+    }
+
+    @Override
+    public void addAssociatedPolicyId(String policyId) {
+        UUID policyUuid = StringKeyConverter.UUIDKey.INSTANCE.fromStringSafe(policyId);
+        if (policyUuid != null) policyIds.add(policyUuid);
```

Comment:
I had a look what `fromStringSafe` does: it will return a null when the `policyId` is not a valid UUID. And then it is not stored to the database. 

As this silently discards data, it is hard to me to see a situation where it is safe to use. Feel free to open a new issue for that.

Reply:
I agree, we should rather throw an exception here. I'll let to throw the original `IllegalArgumentException` here, wdyt?

Reply:
@vramik: +1 for the exception.